frontend/app/routes/_gcweb-app.personal-information.mailing-address.edit.tsx

```diff
@@ -133,67 +180,101 @@ export default function PersonalInformationMailingAddressEdit() {
-            Copy home address
-          </label>
-        </div>
-
-        <InputField
-          id="address"
-          label={t('personal-information:mailing-address.edit.field.address')}
-          name="address"
-          required
-          key={isCopyAddressChecked ? homeAddressInfo?.address : addressInfo?.address}
-          defaultValue={defaultValues.address}
-          errorMessage={errorMessages.address}
-        />
-        <InputField
-          id="city"
-          label={t('personal-information:mailing-address.edit.field.city')}
-          name="city"
-          required
-          key={isCopyAddressChecked ? homeAddressInfo?.city : addressInfo?.city}
-          defaultValue={defaultValues.city}
-          errorMessage={errorMessages.city}
-        />
-        <InputField
-          id="province"
-          label={t('personal-information:mailing-address.edit.field.province')}
-          name="province"
-          key={isCopyAddressChecked ? homeAddressInfo?.province : addressInfo?.province}
-          defaultValue={defaultValues.province}
-          errorMessage={errorMessages.province}
-        />
-        <InputField
-          id="postalCode"
-          label={t('personal-information:mailing-address.edit.field.postal-code')}
-          name="postalCode"
-          key={isCopyAddressChecked ? homeAddressInfo?.postalCode : addressInfo?.postalCode}
-          defaultValue={defaultValues.postalCode}
-          errorMessage={errorMessages.postalCode}
-        />
-        <InputField
-          id="country"
-          label={t('personal-information:mailing-address.edit.field.country')}
-          name="country"
-          required
-          key={isCopyAddressChecked ? homeAddressInfo?.country : addressInfo?.country}
-          defaultValue={defaultValues.country}
-          errorMessage={errorMessages.country}
-        />
-
-        <div className="flex flex-wrap gap-3">
-          <button id="change-button" className="btn btn-primary btn-lg">
-            {t('personal-information:mailing-address.edit.button.change')}
-          </button>
-          <Link id="cancel-button" to="/personal-information" className="btn btn-default btn-lg">
-            {t('personal-information:mailing-address.edit.button.cancel')}
-          </Link>
-        </div>
-      </Form>
+      {isCopyAddressChecked && (
+        <Form className="max-w-prose" method="post">
+          <p>
+            <strong>{t('personal-information:mailing-address.edit.copy-home-address-note')}</strong>
+          </p>
+          <Address
+            address={homeAddressInfo?.address ?? ''}
+            city={homeAddressInfo?.city ?? ''}
+            provinceState={regionList.find((region) => region.code === homeAddressInfo?.province)?.nameEn}
+            postalZipCode={homeAddressInfo?.postalCode ?? ''}
+            country={countryList.find((country) => country.code === homeAddressInfo?.country)?.nameEn ?? ' '}
+          />
```

Comment:
We should do a check to see if the `homeAddressInfo` exists along side the `isCopyAddressChecked` check. That way, the user won't have that option of copying their home address if there is none on file. We would be able to remove all those optional chaining (ie. `homeAddressInfo?.address` becomes `homeAddressInfo.address`).

Reply:
fixed